### PR TITLE
fix: Use plain list instead of <FlatList/> to render servers history dropdown

### PR DESCRIPTION
**Pull request by floels**
## Proposed changes

As discussed in issue #4975, the goal of this PR is to switch the rendering of the servers history dropdown from a `<FlatList />` to a plain list of items, `map()`-ing over the `serversHistory` array, so as to get rid of the "VirtualizedLists should never be nested inside plain ScrollViews with the same orientation" console error when tapping into the server input.

This change doesn't have any visible impact.

I used the occasion to:

- standardize the pluralization to `serversHistory` instead of `serverHistory` everywhere in the codebase, to make it more consistent (both pluralizations were used, leading to weird code),
- move all the database querying logic from `NewServerView/index.tsx` down to `NewServerView/ServerInput/index.tsx`, since `ServerInput` is the only component that needs the corresponding data.

Those three changes are made in separate commits.

## Issue(s)	

#4975 (includes reproduction instructions)

### Before

ezgif-2-8fe554e4f5

### After

ezgif-2-acbddfbe83

## Types of changes

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
[N/A] I have added tests that prove my fix is effective or that my feature works (if applicable)
[N/A] I have added necessary documentation (if applicable)
[N/A] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/NewServerView/ServerInput/index.tsx [hunk 1]
```diff
@@ -72,14 +74,15 @@ const ServerInput = ({
...
+					{/*  https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/5aaf7af5f4004e8b0b9386da579ae4c8aed3a9a0/app/views/NewServerView/index.tsx#L160 */}
+					{/* But we repeat the filtering here, since having more than three items would break the display pretty badly. */}
+					{serversHistory.slice(0, MAX_ITEMS_HISTORY).map(item => (
+						<View key={item.id}>
 							<Item item={item} theme={theme} onPress={() => onPressServerHistory(item)} onDelete={onDelete} />
-						)}
-						ItemSeparatorComponent={List.Separator}
-						keyExtractor={item => item.id}
-					/>
+							<Separator />
```
Comment (diegolmello) [inline app/views/NewServerView/ServerInput/index.tsx:83]: This way you're doubling border bottom
image

File: app/views/NewServerView/ServerInput/index.tsx [hunk 2]
```diff
@@ -72,14 +74,15 @@ const ServerInput = ({
 						{ backgroundColor: themes[theme].backgroundColor, borderColor: themes[theme].separatorColor }
 					]}
 				>
-					<FlatList
-						data={serversHistory}
-						renderItem={({ item }) => (
+					{/* NB: in theory the .slice(0, MAX_ITEMS_HISTORY) is useless because the number of items in the history is already limited to three when querying the database: */}
+					{/*  https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/5aaf7af5f4004e8b0b9386da579ae4c8aed3a9a0/app/views/NewServerView/index.tsx#L160 */}
+					{/* But we repeat the filtering here, since having more than three items would break the display pretty badly. */}
```
Comment (diegolmello) [inline app/views/NewServerView/ServerInput/index.tsx:79]: You have a point.
That's a weird colocation imo.
What if you write a custom hook to fetch servers history?
You could remove it from NewServerView/index and place it here only.
It only matters here really.

The max items logic would be only inside the custom hook.
wdyt?

**Review by diegolmello — CHANGES_REQUESTED** [review]

**Comment by diegolmello** [comment]
@floels btw we just started to use conventional commits on the company.

File: app/views/NewServerView/ServerInput/index.tsx (hunk 2 above)
Comment (floels) [reply app/views/NewServerView/ServerInput/index.tsx]: @diegolmello true. So you would move all the logic around the servers history down in the `<ServerInput />` component?

Comment (floels) [reply app/views/NewServerView/ServerInput/index.tsx]: @diegolmello I'll give it a try on this PR. I'll add a separate commit.

**Comment by floels** [comment]
ℹ️ Keeping this as draft while solving my issue with the local Android build (#4999)

File: app/views/NewServerView/ServerInput/index.tsx (hunk 2 above)
Comment (floels) [reply app/views/NewServerView/ServerInput/index.tsx]: Hi @diegolmello,

I moved the logic to the <ServerInput /> component, as discussed. The PR is ready for a new review.

**Comment by floels** [comment]
> @floels btw we just started to use conventional commits on the company. Also make sure to run e2e tests on your machine.

@diegolmello noted for the conventional commits.

I wasn't able to run E2E tests on my machine unfortunately, the E2E build fails on both iOS and Android 😕 

Let's see what the CI says? 🙈

**Pull request closed without merging**